File: frontend/src/types/tare.ts
export type TareStatus = "inbound" | "storage" | "picking" | "outbound" | "closed";

export type TareType = {
  id: number;
  code: string;
  name: string;
  prefix: string;
  level: number;
};

export type Tare = {
  id: number;
  warehouse_id: number;
  type_id: number;
  location_id: number | null;
  parent_tare_id: number | null;
  tare_code: string;
  status: TareStatus;
  created_at?: string | null;
  updated_at?: string | null;
};

export type TareItemWithItem = {
  id: number;
  tare_id: number;
  item_id: number;
  quantity: number;
  item_sku: string;
  item_name: string;
  item_unit: string;
};

File: backend/app/schemas/tare.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class TareTypeCreate(BaseModel):
    code: str
    name: str
    prefix: str
    level: int = 1


class TareTypeRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    code: str
    name: str
    prefix: str
    level: int


class TareCreate(BaseModel):
    warehouse_id: int
    type_id: int
    location_id: Optional[int] = None
    parent_tare_id: Optional[int] = None


class TareRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    warehouse_id: int
    type_id: int
    location_id: Optional[int] = None
    parent_tare_id: Optional[int] = None
    tare_code: str
    status: str
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None


class TareItemRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    tare_id: int
    item_id: int
    quantity: int
